django/contrib/contenttypes/management/commands/remove_stale_contenttypes.py

```diff
@@ -22,13 +26,18 @@ def add_arguments(self, parser):
     def handle(self, **options):
         db = options['database']
         interactive = options['interactive']
+        include_stale_apps = options['include_stale_apps']
         verbosity = options['verbosity']
 
-        for app_config in apps.get_app_configs():
-            content_types, app_models = get_contenttypes_and_models(app_config, db, ContentType)
+        content_types_by_app = groupby(ContentType.objects.using(db).order_by('app_label', 'model'),
```

Comment:
We should keep a fast return (see `get_contenttypes_and_models()`) when `router` doesn't allow for migrations and clear cache, e.g.
```python
if not router.allow_migrate_model(db, ContentType):
    return
ContentType.objects.clear_cache()
```